api/routers/export.py

```diff
@@ -0,0 +1,169 @@
+    if not os.path.exists(source_path):
+        return None
+    for filename in os.listdir(source_path):
+        if filename.endswith(".csv"):
+            if from_date and to_date:
+                # Date range was specified, only include files in that range
+                date_matches = re.findall(r"([0-9]{4}\-[0-9]{2}\-[0-9]{2})", filename)
+                if not date_matches:
+                    continue
+                file_date = datetime.strptime(date_matches[0], '%Y-%m-%d').date()
+                if not from_date <= file_date <= to_date:
+                    continue
+            zip_file.write(os.path.join(source_path, filename), arcname=os.path.join(destination_path, filename))
+
+
+def get_areas(export_info: ExportDTO) -> List[Tuple[str, str]]:
+    all_areas = extract_config("areas").values()
+    selected_areas = []
+    if export_info.all_areas:
+        selected_areas = all_areas
+    else:
+        selected_areas = [a for a in all_areas if a["Id"] in export_info.areas]
+        if len(selected_areas) != len(export_info.areas):
+            # Some of the selected areas don't exit
+            missing_areas = set(export_info.areas) - set([a["Id"]for a in selected_areas])
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Areas with ids {missing_areas} don't exist."
+            )
+    if selected_areas:
+        return [(area["Id"], area["Name"]) for area in selected_areas]
+    return []
+
+
+def get_cameras(export_info: ExportDTO, areas: List[Tuple[str, str]]) -> List[Tuple[str, str]]:
+    all_cameras = extract_config("cameras").values()
+    selected_cameras = []
+    if export_info.all_cameras:
+        selected_cameras = all_cameras
+    else:
+        selected_cameras = [c for c in all_cameras if c["Id"] in export_info.cameras]
+        if len(selected_cameras) != len(export_info.cameras):
+            # Some of the selected cameras don't exit
+            missing_cameras = set(export_info.cameras) - set([c["Id"]for c in selected_cameras])
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Cameras with ids {missing_cameras} don't exist."
+            )
+        # Include areas' cameras
+        areas_cameras = []
+        areas_ids = [a[0] for a in areas]
+        for area in [a for a in extract_config("areas").values() if a['Id'] in areas_ids]:
+            areas_cameras.extend(area["Cameras"].split(","))
+        selected_cameras.extend(
+            [c for c in all_cameras if c["Id"] in areas_cameras and c["Id"] not in export_info.cameras]
+        )
+    if selected_cameras:
+        return [(camera["Id"], camera["Name"]) for camera in selected_cameras]
+    return []
+
+
+def export_camera_data_into_file(export_info: ExportDTO, camera_id: str, camera_name: str, zip_file: str) -> None:
+    if ALL_DATA in export_info.data_types or RAW_DATA in export_info.data_types:
```

Comment:
I may be mistaken, but shouldn't inclusion be validated in the opposite way?

Reply:
:thinking: What do you mean by "the opposite way"?

Reply:
Nevermind, I just reread the code and understood the constants were strings :upside_down_face: I don't know why yesterday I thought they were lists or something, maybe the same reason as my previous comment :sweat_smile: